libclc/amdgcn-amdhsa/libspirv/async/wait_group_events.cl

```diff
@@ -0,0 +1,17 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <spirv/spirv.h>
+
+_CLC_DEF void _Z23__spirv_GroupWaitEventsjiP9ocl_event(unsigned int scope,
+                                                    int num_events,
+                                                    event_t __attribute__((address_space(0)))* event_list) {
```

Comment:
I think using `__private` is better here.
OTOH, I don't fully understand the problem. If we compile this code with -cl-std=CL2.0, `event_list` will point to generic address space.
What is exactly is the problem here?

Reply:
`0` is generic here (`5` is private)

> If we compile this code with -cl-std=CL2.0

there is a need for selective enabling of 2.0. The lib is tailored for 1.2 so that will probably not turned as expected (you suddenly make pointers to private pointers to generic).

Reply:
Oh... wait. Are we calling OpenCL 1.2 built-in here? Shouldn't we apply private address space attribute on SYCL side in this case and explicitly cast to correct AS?

Reply:
Well there is an overload for generic. Never mattered for PTX because the target address space for private and generic overlap.

The runtime header could be patch to call the privare overload (it will always be a pointer to private AFAIK).

But TBH, there is an issue in the device event design as the stored pointer is a dangling one https://github.com/intel/llvm/issues/4959

Reply:
> The runtime header could be patch to call the privare overload (it will always be a pointer to private AFAIK).

So, which way do you think is better to handle this issue: update the runtime to call correct overload overload or apply this patch?

Reply:
@Naghasan I think the question above was meant for you.

Reply:
oups sorry

> So, which way do you think is better to handle this issue: update the runtime to call correct overload overload or apply this patch?

There is a need for a broader course of action regarding generic address space handling here, so as the patch is ready, I would say ship it. Once the generic is better handled, this will go away.